assets/wizards/site-design/index.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Site Design
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { Component, render, Fragment } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Material UI dependencies.
+ */
+import HeaderIcon from '@material-ui/icons/Web';
+
+/**
+ * Internal dependencies.
+ */
+import { withWizard } from '../../components/src';
+import Router from '../../components/src/proxied-imports/router'
+import { ThemeSelection } from './views';
+
+const { HashRouter, Redirect, Route, Switch } = Router;
+
+class SiteDesignWizard extends Component {
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			theme: null,
+		};
+	}
+
+	componentDidMount = () => {
+		this.retrieveTheme();
+	};
+
+	retrieveTheme = () => {
+		const { setError, wizardApiFetch } = this.props;
+		const params = { path: '/newspack/v1/wizard/newspack-setup-wizard/theme', method: 'GET' };
+		return new Promise( ( resolve, reject ) => {
```

Comment:
Since `retrieveTheme` is only being called when the component mounts, don't really need to wrap it in a promise. Could be simplified to this:

```
wizardApiFetch( params )
			.then( response => {
				const { theme } = response;
				this.setState( { theme } );
			} )
			.catch( error => {
				console.log( '[Theme Fetch Error]', error );
				setError( { error } );
			} );
```

Reply:
Done in https://github.com/Automattic/newspack-plugin/pull/421/commits/7ecda47771051eac8754357f6992a50e17224561